### PR TITLE
fix(common): fix conversion between Decimal to arrow Decimal128

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Previously the conversion is wrong, which will drop the decimal part. This pr fix it and use DECIMAL_DEFAULT_SCALE=10 for 
 `Decimal128`. Previous issue will affect parquet file source and sink. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by wcy-fdu** [comment]
There is no fixed standard for how many decimal places to use, I see [`IcebergCreateTableArrowConvert`](https://github.com/risingwavelabs/risingwave/blob/c0d6af170770ca34635c467ac03f5ac8327c9b21/src/common/src/array/arrow/arrow_iceberg.rs#L182) uses 10 here, which is consistent.

File: src/common/src/array/arrow/arrow_impl.rs [hunk 1]
```diff
@@ -413,10 +413,12 @@ pub trait ToArrow {
 
     #[inline]
     fn decimal_type_to_arrow(&self, name: &str) -> arrow_schema::Field {
-        arrow_schema::Field::new(name, arrow_schema::DataType::Utf8, true)
-            .with_metadata([("ARROW:extension:name".into(), "arrowudf.decimal".into())].into())
```
Comment (chenzl25) [inline src/common/src/array/arrow/arrow_impl.rs:417]: I remember UDF will encode decimal to utf8. This is a breaking change 🥵

Comment (wcy-fdu) [reply src/common/src/array/arrow/arrow_impl.rs:417]: I thought it was a bug, so let's do not change the arrowudf part, but only changed the first place.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**